pkgs/by-name/pa/paper-plane/tdlib.nix

```diff
@@ -0,0 +1,40 @@
+{ fetchFromGitHub, gperf, openssl, readline, zlib, cmake, lib, stdenv }:
+
+stdenv.mkDerivation {
+  pname = "tdlib-paperplane";
+  version = "1.8.19";
+
+  src = fetchFromGitHub {
+    owner = "tdlib";
+    repo = "td";
+
+    # The tdlib authors do not set tags for minor versions, but
+    # external programs depending on tdlib constrain the minor
+    # version, hence we set a specific commit with a known version.
+    rev = "2589c3fd46925f5d57e4ec79233cd1bd0f5d0c09";
+    hash = "sha256-mbhxuJjrV3nC8Ja7N0WWF9ByHovJLmoLLuuzoU4khjU=";
+  };
+
+  buildInputs = [ gperf openssl readline zlib ];
+  nativeBuildInputs = [ cmake ];
+
+  # https://github.com/tdlib/td/issues/1974
+  postPatch = ''
+    substituteInPlace CMake/GeneratePkgConfig.cmake \
+      --replace 'function(generate_pkgconfig' \
```

Comment:
replace-fail is proffered

Reply:
It's just an old revision of tdlib copied from git history, don't expect too much from it

Reply:
`replace-fail` is really optional as its very recent change, i would't want to hold this PR up